File: TUM_Lens/app/src/main/java/com/maxjokel/lens/helpers/ResultItemComparator.java
package com.maxjokel.lens.helpers;

import java.util.Comparator;

public class ResultItemComparator implements Comparator<ResultItem> {
    @Override
    public int compare(ResultItem r1, ResultItem r2) {

        // IDEA:
        // if the number of 'occurrences' is equal, then take a look at 'confidence'

        // TODO
        // this approach uses the 'confidence sum'
        // this might affect the displayed sort order, as we divide the sum by the number of occurrences...

        int t = r2.getOccurrences() - r1.getOccurrences();

        if(t == 0){ // occurrences do not differ

            if (r1.getConfidence() > r2.getConfidence()) return -1;
            if (r1.getConfidence() < r2.getConfidence()) return 1;
            return 0;

        } else { // occurrences differ
            return t;
        }

    }
}

File: TUM_Lens/app/src/main/java/com/maxjokel/lens/helpers/ResultItemComparator.kt
package com.maxjokel.lens.helpers

import java.util.*

class ResultItemComparator : Comparator<ResultItem> {

    override fun compare(r1: ResultItem, r2: ResultItem): Int {
        // IDEA: if the number of 'occurrences' is equal, then take a look at 'confidence'
        // TODO: this approach uses the 'confidence sum' and this might affect the
        //       displayed sort order, as we divide the sum by the number of occurrences...
        val t = r2.occurrences - r1.occurrences
        if (t == 0) { // occurrences do not differ
            if (r1.confidence > r2.confidence) return -1
            if (r1.confidence < r2.confidence) return 1 else return 0
        } else { // occurrences differ
            return t
        }
    }
}